### PR TITLE
CDAP-19660 - Encode key for app state to avoid malforming the remote URL.

### PR DESCRIPTION
**Pull request by greeshmaswaminathan**
Encode key for app state to avoid malforming the remote URL.

### CODE REVIEW

File: cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/runtime/RemoteAppStateStoreTest.java [hunk 1]
```diff
@@ -98,95 +179,88 @@ public static void cleanUp() throws Exception {
...
   }
 
   @Test
   public void testSaveInvalidNamespace() throws IOException {
     expectedException.expectCause(CoreMatchers.isA(NotFoundException.class));
-    RemoteAppStateStore remoteAppStateStore = new RemoteAppStateStore(cConf, remoteClientFactory, "invalid", "app1");
-    byte[] value = "testvalue".getBytes();
-    remoteAppStateStore.saveState("key1", value);
+    RemoteAppStateStore remoteAppStateStore = new RemoteAppStateStore(cConf, remoteClientFactory,
+                                                                      "invalid", "some_app");
```
Comment (snehalakshmisha) [inline cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/runtime/RemoteAppStateStoreTest.java:198]: nit: indentation

File: cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/runtime/RemoteAppStateStoreTest.java [hunk 2]
```diff
@@ -98,95 +179,88 @@ public static void cleanUp() throws Exception {
...
   }
 
   @Test
-  public void testGetSuccess() throws ApplicationNotFoundException, IOException {
-    RemoteAppStateStore remoteAppStateStore = new RemoteAppStateStore(cConf, remoteClientFactory, NAMESPACE, "app1");
-    byte[] value = "testvalue".getBytes();
-    Mockito.when(applicationLifecycleService.getState(Mockito.any())).thenReturn(Optional.of(value));
-    Optional<byte[]> state = remoteAppStateStore.getState("key1");
-    Assert.assertEquals(new String(value), new String(state.get()));
+  public void getWithValidKeys() throws IOException {
```
Comment (snehalakshmisha) [inline cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/runtime/RemoteAppStateStoreTest.java]: nit: start method with `test`

**Review by snehalakshmisha — APPROVED** [review]
Can you please add a jira for this?

**Comment by greeshmaswaminathan** [comment]
> Can you please add a jira for this?

JIRA is already added in the title - CDAP-19660

File: cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/runtime/RemoteAppStateStoreTest.java (hunk 1 above)
Comment (greeshmaswaminathan) [reply cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/runtime/RemoteAppStateStoreTest.java:198]: Indentation matches with the method params.

File: cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/runtime/RemoteAppStateStoreTest.java (hunk 2 above)
Comment (greeshmaswaminathan) [reply cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/runtime/RemoteAppStateStoreTest.java]: Done

**Pull request merged**